pkg/endpointmanager/host.go

```diff
@@ -36,20 +38,7 @@ func (mgr *endpointManager) HostEndpointExists() bool {
 // This adheres to the subscriber.NodeHandler interface.
 func (mgr *endpointManager) OnAddNode(newNode *slim_corev1.Node,
 	swg *lock.StoppableWaitGroup) error {
-
-	node.SetLabels(newNode.GetLabels())
-
-	nodeEP := mgr.GetHostEndpoint()
-	if nodeEP == nil {
-		// if host endpoint does not exist yet, labels will be set when it'll be created.
-		return nil
-	}
-
-	newLabels := labels.Map2Labels(newNode.GetLabels(), labels.LabelSourceK8s)
-	newIdtyLabels, _ := labelsfilter.Filter(newLabels)
-	nodeEP.UpdateLabels(context.TODO(), labels.LabelSourceAny, newIdtyLabels, nil, false)
-
-	return nil
+	return mgr.OnUpdateNode(&slim_corev1.Node{}, newNode, swg)
```

Comment:
nit: pass `nil` here, handle nil below. Otherwise, this looks a lot better.

Reply:
Done